### PR TITLE
#291 added tests for reservation settings

### PR DESCRIPTION
**Pull request by krzyk**
#291 
* added tests for reservation settings
* todo to tune and enable the tests when the functionality will be ready

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#335](https://github.com/timelesslounge/timelessis/issues/335) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#335](https://github.com/timelesslounge/timelessis/issues/335) is assigned to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/4b50f7c3-b938-456d-a42f-281f3ed5258f); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by krzyk** [comment]
@vladarefiev ping

File: tests/integration/it_reservation_settings_test.py [hunk 1]
```diff
@@ -1,36 +1,81 @@
...
+    data = reservation_data("nicer comment here")
+    new_data = reservation_data("different comment here")
+    client.post(url_for("reservations.settings_create"), data=data)
+    identifier = models.ReservationSettings.query.first().id
+    client.post(
+        url_for("reservations.settings_create", id=identifier), data=new_data
+    )
+    response = client.get(url_for("reservations.settings_list"))
+    assert response.status_code == HTTPStatus.OK
+    html = response.data.decode('utf-8')
```
Comment (vladarefiev) [inline tests/integration/it_reservation_settings_test.py]: @krzyk single quoutes

File: tests/integration/it_reservation_settings_test.py [hunk 2]
```diff
@@ -1,36 +1,81 @@
...
+    data = reservation_data("my very unique comment")
+    client.post(url_for("reservations.settings_create"), data=data)
+    identifier = models.ReservationSettings.query.first().id
+    client.post(
+        url_for("reservations.settings_delete"),
+        data={"id": identifier}
+    )
+    response = client.get(url_for("reservations.settings_list"))
+    assert response.status_code == HTTPStatus.OK
+    html = response.data.decode('utf-8')
```
Comment (vladarefiev) [inline tests/integration/it_reservation_settings_test.py]: @krzyk the same

**Comment by vladarefiev** [comment]
@krzyk Really good to me. Only one comment.

**Comment by krzyk** [comment]
@vladarefiev Updated

**Comment by vladarefiev** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 18 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@fabb3db5-c392-4586-bd85-c749a1dd1755, thread: pool-3-thread-6
org.xembly.Xembler[132] java.lang.IllegalArgumentException: failed to apply to DOM quietly: XPATH "/agenda/order[@job=&apos;gh:timelesslounge/timelessis#335&apos;]";STRICT "1";
2:ADDIF "inspector";SET "ypshenychka";
org.xembly.Xembler[158] org.xembly.ImpossibleModificationException: directive #2: STRICT "1"
org.xembly.StrictDirective[72] org.xembly.ImpossibleModificationException: no current nodes while 1 expected
```

[0.42.1](https://github.com/zerocracy/farm): CID: [0a0fce2e-188f-4cb8-9d27-d4f99f63b373](https://www.0crat.com/footprint/CF9LG1YCA/0a0fce2e-188f-4cb8-9d27-d4f99f63b373), Type: "Start QA review"

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/timelesslounge/timelessis/issues/335#issuecomment-466183640))

@ypshenychka This is what I know about this job in [CF9LG1YCA](https://www.0crat.com/p/CF9LG1YCA), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#335](https://github.com/timelesslounge/timelessis/issues/335) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 12hr
  * @vladarefiev[/z](https://www.0crat.com/u/vladarefiev) will receive 20 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/CF9LG1YCA?q=%7Bjob%3A%27gh%3Atimelesslounge%2Ftimelessis%23335%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@vladarefiev  According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by vladarefiev** [comment]
@ypshenychka confirm

**Comment by ypshenychka** [comment]
@vladarefiev thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +20 point(s) just awarded to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**